gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/v4/impl/ApiServiceImpl.java

```diff
@@ -466,8 +471,8 @@ public ApiEntity update(
             if (updateApiEntity.getLabels() == null && apiToUpdate.getLabels() != null) {
                 api.setLabels(new ArrayList<>(new HashSet<>(apiToUpdate.getLabels())));
             }
-            if (updateApiEntity.getCategories() == null) {
-                api.setCategories(apiToUpdate.getCategories());
+            if (updateApiEntity.getCategories() == null) { //!!!
+                api.setCategories(categoryMapper.toCategoryKey(executionContext.getEnvironmentId(), apiToUpdate.getCategories()));
```

Comment:
This should be `toCategoryId` right? It's setting the category list of a repository Api ?

Reply:
No, because in this `update` we do category transformation twice:
`Api api = apiMapper.toRepository(executionContext, updateApiEntity);` does to categoryId
But for updateApiEntity we do to categoryKey, since updateApiEntity will be used in REST response
`api.setCategories(categoryMapper.toCategoryKey(executionContext.getEnvironmentId(), apiToUpdate.getCategories()));`
<img width="1117" alt="Screenshot 2024-05-24 at 09 58 39" src="https://github.com/gravitee-io/gravitee-api-management/assets/8080303/59d2b273-8d83-454a-b692-069237b2aba1">

Reply:
As discussed, the mapper can be removed from line 475 and be replaced with a simple set